components/TextFields/src/private/MDCTextInputCommonFundament.m

```diff
@@ -438,7 +438,7 @@ - (void)subscribeForKVO {
 }
 
 - (void)unsubscribeFromKVO {
-  if (!self.textInput || !self.isRegisteredForKVO) {
+  if (!self.underline || !self.isRegisteredForKVO) {
```

Comment:
The subscription only registers for KVO if self.textInput is non nil. Which of the two is incorrect?